### PR TITLE
Issue parser

### PR DESCRIPTION
**Pull request by lady-ann**

### CODE REVIEW

File: src/main/kotlin/io/easybreezy/project/application/project/queryobject/Settings.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            .join(Users, JoinType.INNER, Members.user, Users.id)
+            .join(Teams, JoinType.INNER, Members.team, Teams.id)
+            .select {
+                Teams.project eq project and (Users.username.lowerCase() inList usernames)
+            }
+            .map { it[Users.id].value }
+    }
+}
+
+class LastLowestPriorityQO(private val project: UUID) : QueryObject<Int?> {
```
Comment (lady-ann) [inline src/main/kotlin/io/easybreezy/project/application/project/queryobject/Settings.kt:52]: I need help to write this simple query with exposed

File: src/main/kotlin/io/easybreezy/project/application/issue/command/parser/PriorityConverter.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import io.easybreezy.project.application.project.queryobject.LastLowestPriorityQO
+import io.easybreezy.project.model.issue.Priority
+import java.util.UUID
+
+class PriorityConverter @Inject constructor(
+    private val queryExecutor: QueryExecutor
+) {
+    suspend fun convert(project: UUID, title: String?): Priority {
+        return when (title?.toLowerCase()) {
+            "high" -> Priority.high()
```
Comment (shustrik) [inline src/main/kotlin/io/easybreezy/project/application/issue/command/parser/PriorityConverter.kt:15]: We should add the ability to use many languages. May create a file with map x keywords and translation

File: src/main/kotlin/io/easybreezy/project/model/issue/Priority.kt [hunk 3]
```diff
@@ -43,16 +43,26 @@ class Priority private constructor() : Embeddable() {
             return priority
         }
 
-        fun highest(lastHighest: Int): Priority {
+        fun highest(lastHighest: Int?): Priority {
             val priority = Priority()
```
Comment (shustrik) [inline src/main/kotlin/io/easybreezy/project/model/issue/Priority.kt:47]: Hmm, As I see every new issue with the highest priority will be on top on the board? What I should do when I need to change the position of these issues?

Comment (lady-ann) [reply src/main/kotlin/io/easybreezy/project/model/issue/Priority.kt:47]: Yes, by adding a comment "the highest priority" to the highest
one

**Comment by shustrik** [comment]
After parsing text we have parse object, my proposal is rename parse to ParsedIssue this issue contains only simple data as a list of string users, string priority and etc. After that  we'll add normalizer which create NormalizerIssues with actual data as a list of UUID's of users, normalized date, priority object, I think for each type of data we should have independent normalizer like UserNormalizer, PriorityNormalizer

File: src/main/kotlin/io/easybreezy/project/application/issue/command/parser/Normalizer.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package io.easybreezy.project.application.issue.command.parser
+
+import com.google.inject.Inject
+import io.easybreezy.project.model.issue.Label
+import io.easybreezy.project.model.issue.Priority
+import java.util.UUID
+
+class Normalizer @Inject constructor(
+    private val categoryConverter: CategoryConverter,
```
Comment (shustrik) [inline src/main/kotlin/io/easybreezy/project/application/issue/command/parser/Normalizer.kt:9]: will be better create converter interface which will be controlled by generic type like CategoryConverter: Converter<List<UUID>?> also we add common function convert(project: UUID, parserIssue: ParsedIssue):  T.
All converters will resolve only self data. Now it's hard to extend converter, we should always add converter on a constructor. When we add interface we'll can build collection of converters

**Pull request merged**